### ISSUE TITLE
SubscribePendingTransactions old mempool txes

### ISSUE THREAD

**Issue by rpcerror** [ethereum/go-ethereum#26794]
Geth version: `v1.11.2`
OS & Version: Windows 11

#### Expected behaviour
Subscribing to only the **NEW** transaction hashes from ETH mempool

#### Actual behaviour
Subscribing to both **NEW** and **OLD** transaction hashes that are pending for hours/days

#### Steps to reproduce the behaviour
run func: gethclient.SubscribePendingTransactions()

Explanation:
I used function client.TransactionByHash() to convert all the new pending tx hashes to *types.Transacton to scrape some info from them. I noticed getting rate limited immidiately. My RPC has limit of 330req/s, and the ETH TPS is currently around 11/12, so I was totally surprised. Then I printed out the subscribed tx hashes and manually checked them on Etherscan. Immidiatelly noticed most of the printed txs are pending since hours/days. From my understanding the function should only be subscribing to **NEW** txes, since the start of the subscription, not printing all pending txs including the old ones from the mempool. Any fix/explanation please? Maybe I am doing something wrong? Using context.Background().

**Comment by rjl493456442** [comment]
Is your node already synced? Or it's still syncing. Could you please check the status via console `eth.syncing`?

**Comment by rjl493456442** [comment]
> My RPC has limit of 330req/s

Are you using some service like Infura? Or you are using the local Geth node.

**Comment by rjl493456442** [comment]
Yeah, You may want to setup your own geth node. And unfortunately, if it's some external service, we can't do too much here. They can have some additional logics and might not be relevant with Geth.

**Comment by karalabe** [comment]
The network TPS is not really relevant for subscribing to pending transactions. People are non stop submitting transactions, replacing their own to bump the price limit a bit. The txpool sees significantly more "intermediate" transactions than what end up in the final block. By subscribing to pending transactions, you should get the entire stream of txs that are added to the pool, even those that will end up being dropped seconds later. This seems to be logical to me.

**Comment by karalabe** [comment]
If you think this is still wrong and it delivers junk it should not, perhaps can you create a stream of transactions you're seeing and submit it in some logs so we might check it?

**Comment by holiman** [comment]
Example: https://etherscan.io/tx/0x6a06379d7992f49083b1c08284b2276f3bb36a2611042bfc7763bb10e29bbb7c  -- this one is still pending. It's a very low-priced transaction, so will likely be discarded by the pool at some point, and then possibly re-added again later, and wind up in `pending` again.

**Comment by fjl** [comment]
Here's a theory how txs might enter the 'pending' area multiple times: When a transaction is replaced by another one with a higher fee, we move the remaining txs of the removed tx's sender account into 'queued'. If we then get another tx from the same account, those queued txs will re-enter 'pending' and will be delivered on the subscription another time.

I'm pretty sure the behavior observed here is related to activity at the tail end of the fee market. A lot of replacing/queueing will happen there.
